lms/djangoapps/support/views/course_entitlements.py

```diff
@@ -0,0 +1,133 @@
+                    )
+                )
+            if entitlement.enrollment_course_run is None:
+                return HttpResponseBadRequest(
+                    u"Entitlement for user {user} to course {course} has not been spent on a course run.".format(
+                        user=entitlement.user.username,
+                        course=entitlement.course_uuid
+                    )
+                )
+            unenrolled_run = self.unexpire_entitlement(entitlement)
+            CourseEntitlementSupportDetail.objects.create(
+                entitlement=entitlement, reason=reason, comments=comments, unenrolled_run=unenrolled_run,
+                support_user=support_user
+            )
+            return Response(
+                status=status.HTTP_201_CREATED,
+                data=SupportCourseEntitlementSerializer(instance=entitlement).data
+            )
+        else:
+            return HttpResponseBadRequest(
+                u'Could not find an entitlement for user {username} in course {course}'.format(
+                    username=username_or_email,
+                    course=course_uuid
+                )
+            )
+
+    @method_decorator(require_support_permission)
+    def create(self, request, *args, **kwargs):
+        """ Allows support staff to grant a user a new entitlement for a course. """
+        support_user = request.user
+        try:
+            username_or_email = self.request.data['user']
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            course_uuid = request.data['course_uuid']
+            reason = request.data['reason']
+            comments = request.data.get('comments', None)
+            mode = request.data.get('mode', 'verified')
+        except KeyError as err:
+            return HttpResponseBadRequest(u'The field {} is required.'.format(text_type(err)))
+        except User.DoesNotExist:
+            return HttpResponseBadRequest(
+                u'Could not find user {username}.'.format(
+                    username=username_or_email,
+                )
+            )
+        entitlement = CourseEntitlement.objects.get_or_create(user=user, course_uuid=course_uuid, mode=mode)
+        CourseEntitlementSupportDetail.objects.create(
+            entitlement=entitlement, reason=reason, comments=comments, support_user=support_user
+        )
+        return Response(
+            status=status.HTTP_201_CREATED,
+            data=SupportCourseEntitlementSerializer(instance=entitlement).data
+        )
+
+    @staticmethod
+    def unexpire_entitlement(entitlement):
+        """
+        Unenrolls a user from the run on which they have spent the given entitlement and
+        sets the entitlement's expired_at date to null.
+        """
+        unenrolled_run = entitlement.enrollment_course_run.course
+        entitlement.expired_at = None
+        entitlement.enrollment_course_run.unenroll(skip_refund=True)
```

Comment:
@staubina I'm changing this back to deactivate because for unenroll I have to actually pass the parameters to select the record, rather than performing the action on the enrollment itself (which deactivate does) and it functions essentially the same was as unenroll(skip_refund), which is to set the record to inactive

Reply:
@McKenzieW are there any other hooks that differ between unenroll and deactivate? I.e. analytics (not sure we'd want to necessarily track) or any other signals that get fired?

Reply:
No, they both call the `update_enrollment()` method on the enrollment, which emits an event when activation is changed that would would be triggered by either method

Reply:
The issue here is that deactivate currently calls update_enrollment with the default `skip_refund=False` which will cause the refund signal to trigger.